builder/image/slugbuilder/builder/install-buildpacks

```diff
@@ -8,10 +8,17 @@ download_buildpack() {
     buildpack_name=$(basename $buildpack_url)
     buildpack_commit="$2"
 
-    git clone $buildpack_url $BUILDPACK_INSTALL_PATH/$buildpack_name
-    pushd $BUILDPACK_INSTALL_PATH/$buildpack_name 2>&1 >/dev/null
-        git checkout --quiet $buildpack_commit
-    popd 2>&1 >/dev/null
+    set +e
+    git clone --branch $buildpack_commit --depth 1 $buildpack_url $BUILDPACK_INSTALL_PATH/$buildpack_name
+    SHALLOW_CLONED=$?
+    set -e
+    if [ $SHALLOW_CLONED -ne 0 ]; then
+        git clone $buildpack_url $BUILDPACK_INSTALL_PATH/$buildpack_name
```

Comment:
yes, I believe it would. You're not allowed to use a directory name which already exists, so if it failed partway through that could happen